ecommerce/qverse_features/paystack/client.py

```diff
@@ -0,0 +1,128 @@
+import json
+import logging
+
+import requests
+
+from ecommerce.qverse_features.paystack import constants as paystack_const
+from ecommerce.qverse_features.paystack.exceptions import (
+    InvalidClientArgument,
+    InvalidPaystackClientMethod,
+    InvalidRequestMethod,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class PaystackClient(object):
+    """
+    Client for Paystack API requests
+    """
+
+    _CONTENT_TYPE = "application/json"
+
+    def __init__(self, base_url, authorization_key=None):
+        if base_url and authorization_key:
+            self._BASE_END_POINT = base_url
+            self._AUTHORIZATION_KEY = authorization_key
+        else:
+            msg = "Authorization key" if not authorization_key else "Base Url"
+            raise InvalidClientArgument("Missing {} argument".format(msg))
+
+    def _url(self, path):
+        url = self._BASE_END_POINT
+        if path:
+            url += path
+        return url
+
+    def _headers(self):
+        return {
+            "Content-Type": self._CONTENT_TYPE,
+            "Authorization": "Bearer " + self._AUTHORIZATION_KEY
+        }
+
+    def _parse_response(self, response):
+        try:
+            data = response.json()
+            logger.info(
+                "\nPaystack status: %s, \n Paystack message:%s", data.get('status'), data.get('message')
+            )
+        except:  # pylint: disable=bare-except
+            data = None
+
+        if response.status_code in [200, 201]:
+            logger.info("Paystack API return success response: %s", data)
+            return True, data
+
+        else:
+            logger.error("Paystack API return response with status code: %s", response.status_code)
+            logger.error("Paystack API return Error response: %s", json.dumps(data))
+            return False, data
+
+    def _handle_request(self, request_data):
+        """
+        Generic function to handle all Paystack API calls
+        """
+        method = request_data.get('method')
+        path = request_data.get('path')
+        data = request_data.get('data')
+
+        method_map = {
+            paystack_const.GET_METHOD: requests.get,
```

Comment:
We can make constants for "GET" and "POST" inside our client. 
nit. Just a suggestion since HTTP related constants should be inside the client.